### ISSUE TITLE
Comment imports to tolerate exceptions

### ISSUE THREAD

**Issue by nnutter** [fatih/faillint#6]
Imagine wanting to avoid a package but, in practice, needing to be able to tolerate exceptions to it (while using `faillint` in a CI pipeline).  Much like many linting tools I wonder if the import could be annotated like,

```go
import (
    "github.com/gogo/protobuf/proto" // faillint: gogo is tolerated here because XYZ.
)
```

So if we had `-path "github.com/gogo/protobuf/proto=github.com/golang/protobuf/proto"` but `faillint` saw a `// faillint:` annotation it would suppress its objection (and if no objections exit zero).

**Comment by nnutter** [comment]
If this idea is acceptable I would be happy to explore if it's possible and put together a PR.

**Comment by fatih** [comment]
Thanks nnuter. This makes sense, maybe there are some packages or files where people can't change and have to use certain packages. Do you know how other linters use this in their code? I think `staticcheck` also has this. We should try to use the same annotation pattern.

**Comment by nnutter** [comment]
`staticcheck` uses:

- `//lint:ignore Check1[,Check2,...,CheckN] reason` for line-based ignores
- `//lint:file-ignore Check1[,Check2,...,CheckN] reason` for whole file ignores

Since `faillint` isn't checking multiple classes of problems it would probably make sense to use `//lint:ignore reason` and `//lint:file-ignore reason`.  Another option might be to use `faillint` as a check name in the `staticcheck` style; `staticcheck` has codes like `ST1019` for the check names.

I guess I'll explore this a bit since you seem interested.

**Comment by nnutter** [comment]
I created a PR.  It was pretty easy to add.  We can see what you think of the code, etc.  Happy to iterate on any feedback or if you want to just take it and run with it that's fine with me too.

**Comment by bwplotka** [comment]
+1 to this request, but we might want to make it work for all declarations: https://github.com/fatih/faillint/pull/8 (:

**Comment by nnutter** [comment]
I guess I'll wait until that PR is merged and then maybe take another pass at it.

**Comment by bwplotka** [comment]
Thanks!

**Comment by nnutter** [comment]
New PR up.

**Comment by fatih** [comment]
This is now closed with #11 Thanks again @nnutter on working on it.

Faillint `v1.3.0` includes these changes.

**Comment by bwplotka** [comment]
:muscle:
